assets/js/index.js

```diff
@@ -2,38 +2,54 @@ const addTaskIcon = document.getElementById("add-task-icon");
 const addTaskInput = document.getElementById("add-task-input");
 const errorMessage = document.getElementById("error-message");
 const backendErrorMessage = document.getElementById("backend-error-message");
+const todoList = document.getElementById("todo-list");
+const filterItemAll = document.getElementById("filter-item-all");
+const filterItemUncompleted = document.getElementById(
+  "filter-item-uncompleted"
+);
+const filterItemCompleted = document.getElementById("filter-item-completed");
+const url = "http://localhost:3030";
 
 async function postTodo(todoText) {
-  const url = "http://localhost:3030/todos";
-
   try {
-    const response = await fetch(url, {
+    const response = await fetch(`${url}/todos`, {
       method: "POST",
       headers: {
         "Content-Type": "application/json",
       },
       body: JSON.stringify({ text: todoText }),
     });
-    const {error} = await response.json();
+    const { error, data } = await response.json();
+
     if (response.status !== 201) {
       errorMessage.innerText = error;
     }
   } catch (error) {
-    backendErrorMessage.innerText = "Something went wrong!";
+      showNetworkError();
   }
 }
 
-addTaskIcon.addEventListener("click", function () {
-  postTodo(addTaskInput.value);
+addTaskIcon.addEventListener("click", async function () {
+  await postTodo(addTaskInput.value);
   clearAddTodoInput();
   clearErrorMessage();
+  const tasks = await getTasks();
+  showTasks(tasks);
+  filterItemAllTasks();
+  filterItemCompletedTasks();
+  filterItemUncompletedTasks();
```

Comment:
Create s new function and call it updateFilterCounts and call these 3 functions in there. And use that function here instead.